src/main/java/com/iota/iri/service/API.java

```diff
@@ -353,7 +354,8 @@ private AbstractResponse process(final String requestString, InetSocketAddress s
     }
 
     private boolean isBelowNewTransactionLimit(InetAddress sourceAddress, int size) {
-        if (newTransactionsLimit == 0) {
+        int newTransactionsLimitInt = newTransactionsLimit.get();
```

Comment:
nit: you might want to specify the order as relaxed here, that might give a minor speed boost (default is seq_cst)